File: mycollab-web-pro/src/main/scala/com/mycollab/pro/module/project/events/GanttEvent.scala
package com.mycollab.pro.module.project.events

import com.mycollab.events.ApplicationEvent
import com.mycollab.module.project.domain.TaskPredecessor
import com.mycollab.pro.module.project.view.assignments.gantt.GanttItemWrapper

/**
  * @author MyCollab Ltd
  * @since 5.1.3
  */
object GanttEvent {

  class ClearGanttItemsNeedUpdate(source: AnyRef) extends ApplicationEvent(source, null) {}

  class UpdateGanttItem(source: AnyRef, data: GanttItemWrapper) extends ApplicationEvent(source, data) {}

  class DeleteGanttItemUpdateToQueue(source: AnyRef, data: GanttItemWrapper) extends ApplicationEvent(source, data) {}

  class AddGanttItemUpdateToQueue(source: AnyRef, data: GanttItemWrapper) extends ApplicationEvent(source, data) {}

  class ModifyPredecessors(source: GanttItemWrapper, predecessors: java.util.List[TaskPredecessor]) extends ApplicationEvent(source, predecessors) {}

}

File: mycollab-web-pro/src/main/java/com/mycollab/pro/module/project/event/GanttEvent.kt
package com.mycollab.pro.module.project.event

import com.mycollab.vaadin.event.ApplicationEvent
import com.mycollab.module.project.domain.TaskPredecessor
import com.mycollab.pro.module.project.view.assignments.gantt.GanttItemWrapper

/**
 * @author MyCollab Ltd
 * @since 6.0.0
 */
class GanttEvent {
    class ClearGanttItemsNeedUpdate(source: Any) : ApplicationEvent(source)

    class UpdateGanttItem(source: Any, val data: GanttItemWrapper) : ApplicationEvent(source)

    class DeleteGanttItemUpdateToQueue(source: Any, val data: GanttItemWrapper) : ApplicationEvent(source)

    class AddGanttItemUpdateToQueue(source: Any, val data: GanttItemWrapper) : ApplicationEvent(source)

    class ModifyPredecessors(source: GanttItemWrapper, val predecessors: List<TaskPredecessor>) : ApplicationEvent(source)
}